src/content_cao.h

```diff
@@ -108,14 +108,22 @@ class GenericCAO : public ClientActiveObject
 	float m_step_distance_counter;
 	u8 m_last_light;
 	bool m_is_visible;
+	f32 m_knockback_timer;
+	bool m_knockback;
```

Comment:
you can remove the bool and replace `if(m_knockback)` with `if(m_knockback_timer > 0.0)`